feat(performance): Add `tracesSampleRate` and `tracesSampler` to JS docs (#2376)
getsentry/sentry-docs: src/platforms/javascript/common/performance/manual.mdx

> ---

Reviewer: Should this page be a subpage of "Enabling"?

title: Capturing Transactions Manually
sidebar_order: 20
description: "Learn how to capture performance data on any action in your app."
---

### Manual Instrumentation

To manually instrument certain regions of your code, you can create transactions to capture them. This is valid for all JavaScript SDKs (both backend and frontend) and works independently of the `Express`, `Http`, and `BrowserTracing` integrations.

```javascript
const transaction = Sentry.startTransaction({ name: "test-transaction" });
const span = transaction.startChild({ op: "functionX" }); // This function returns a Span
// functionCallX
span.finish(); // Remember that only finished spans will be sent with the transaction
transaction.finish(); // Finishing the transaction will send it to Sentry
```

> Here is a different example. If you want to create a transaction for a user interaction on your page, you need to do the following:

Reviewer: ```suggestion
As another example, if you want to create a transaction for a user interaction on your page:
```

```javascript
// Let's say this function is invoked when a user clicks on the checkout button of your shop
shopCheckout() {
  // This will create a new Transaction for you
  const transaction = Sentry.startTransaction('shopCheckout');
  // set the transaction on the scope so it picks up any errors
  hub.configureScope(scope => scope.setSpan(transaction));

  // Assume this function makes an xhr/fetch call
  const result = validateShoppingCartOnServer();

  const span = transaction.startChild({
    data: {
      result
    },
    op: 'task',
    description: `processing shopping cart result`,
  });
  processAndValidateShoppingCart(result);
  span.finish();

  transaction.finish();
}
```

This example will send a transaction `shopCheckout` to Sentry. The transaction will contain a `task` span that measures how long `processAndValidateShoppingCart` took. Finally, the call to `transaction.finish()` will finish the transaction and send it to Sentry.

#### Adding Additional Spans to the Transaction

> The next example contains the implementation of the hypothetical `processItem` function called from the code snippet in the previous section. Our SDK can determine if there is currently an open transaction and add to it all newly created spans as child operations. Keep in mind that each individual span needs to be manually finished; otherwise, that span will not show up in the transaction.

Reviewer: ```suggestion
The next example contains the implementation of the hypothetical `processItem` function called from the code snippet in the previous section. Our SDK determines if there is a current, open transaction and adds to it all newly created spans as child operations. Keep in mind that each individual span needs to be manually finished; otherwise, that span will not show up in the transaction.
```

Author: Now that I look at it, this was clearly left over from a previous iteration of the API, because the text didn't match the code. Rewritten.
